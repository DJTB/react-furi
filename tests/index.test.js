import React from 'react';
import { render } from 'react-testing-library';

import ReactFuri from '../src/index';
const { Wrapper, Pair, Furi, Text } = ReactFuri;

describe('<ReactFuri />', () => {
  it('no reading or furi provided: just render word', () => {
    const { container } = render(<ReactFuri word="漢字" />);
    expect(container).toMatchSnapshot();
  });

  it('no furi provided: render relevant readings over kanji blocks', () => {
    const { container } = render(<ReactFuri word="お見舞い" reading="おみまい" />);
    expect(container).toMatchSnapshot();
  });

  it('furi provided: render furigana over related kanji blocks', () => {
    const { container } = render(<ReactFuri word="漢字" furi="0:かん;1:じ" />);
    expect(container).toMatchSnapshot();
  });

  it('showFuri false: no furigana rendered', () => {
    const { container } = render(<ReactFuri word="漢字" furi="0:かん;1:じ" showFuri={false} />);
    expect(container).toMatchSnapshot();
  });

  it('custom render prop', () => {
    const { container } = render(
      <ReactFuri
        word="漢字"
        furi="0:かん;1:じ"
        render={({ pairs }) => (
          <Wrapper lang="ja">
            {pairs.map(([furiText, text], index) => (
              // eslint-disable-next-line react/no-array-index-key
              <Pair key={index}>
                {furiText && <Furi>{furiText}</Furi>}
                <Text>{text}</Text>
              </Pair>
            ))}
          </Wrapper>
        )}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
