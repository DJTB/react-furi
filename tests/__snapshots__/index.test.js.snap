// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactFuri /> custom render prop 1`] = `
<div>
  <span
    class="sc-bdVaJa hSpgnF"
    lang="ja"
  >
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      >
        かん
      </span>
      <span
        class="sc-bxivhb imCjfa"
      >
        漢
      </span>
    </span>
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      >
        じ
      </span>
      <span
        class="sc-bxivhb imCjfa"
      >
        字
      </span>
    </span>
  </span>
</div>
`;

exports[`<ReactFuri /> furi provided: render furigana over related kanji blocks 1`] = `
<div>
  <span
    class="sc-bdVaJa hSpgnF"
    lang="ja"
  >
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      >
        かん
      </span>
      <span
        class="sc-bxivhb imCjfa"
      >
        漢
      </span>
    </span>
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      >
        じ
      </span>
      <span
        class="sc-bxivhb imCjfa"
      >
        字
      </span>
    </span>
  </span>
</div>
`;

exports[`<ReactFuri /> no furi provided: render relevant readings over kanji blocks 1`] = `
<div>
  <span
    class="sc-bdVaJa hSpgnF"
    lang="ja"
  >
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      />
      <span
        class="sc-bxivhb imCjfa"
      >
        お
      </span>
    </span>
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      >
        みま
      </span>
      <span
        class="sc-bxivhb imCjfa"
      >
        見舞
      </span>
    </span>
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      />
      <span
        class="sc-bxivhb imCjfa"
      >
        い
      </span>
    </span>
  </span>
</div>
`;

exports[`<ReactFuri /> no reading or furi provided: just render word 1`] = `
<div>
  <span
    class="sc-bdVaJa hSpgnF"
    lang="ja"
  >
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-htpNat cKlPSt"
      />
      <span
        class="sc-bxivhb imCjfa"
      >
        漢字
      </span>
    </span>
  </span>
</div>
`;

exports[`<ReactFuri /> showFuri false: no furigana rendered 1`] = `
<div>
  <span
    class="sc-bdVaJa hSpgnF"
    lang="ja"
  >
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-bxivhb imCjfa"
      >
        漢
      </span>
    </span>
    <span
      class="sc-bwzfXH faybqm"
    >
      <span
        class="sc-bxivhb imCjfa"
      >
        字
      </span>
    </span>
  </span>
</div>
`;
